src/plugins/intel_cpu/src/ngraph_transformations/convert_to_interaction.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_to_interaction.hpp"
+#include "op/interaction.hpp"
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/core/rt_info.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "itt.hpp"
+
+ov::intel_cpu::ConvertToInteraction::ConvertToInteraction() {
+    MATCHER_SCOPE(ConvertToInteraction);
+    using namespace ov::pass::pattern;
+    auto dense_feature_m = any_input(has_static_rank());
+    std::vector<std::shared_ptr<Node>> features_m{dense_feature_m};
+    OutputVector features_output{dense_feature_m->output(0)};
+    const int sparse_feature_num = 26;
+    for (size_t i = 0; i < sparse_feature_num; i++) {
+        auto feature = any_input(has_static_rank());
+        features_m.push_back(feature);
+        features_output.push_back(feature->output(0));
+    }
+    auto concat_m = wrap_type<ov::opset8::Concat>(features_output);
+    auto reshape_m = wrap_type<ov::opset8::Reshape>({concat_m->output(0), any_input()->output(0)});
+    auto matmul_m = wrap_type<ov::opset1::MatMul>({reshape_m, reshape_m});
+    auto transpose2_m = wrap_type<ov::opset1::Transpose>({matmul_m->output(0), any_input()->output(0)});
+    auto reshape2_m = wrap_type<ov::opset1::Reshape>({transpose2_m->output(0), any_input()->output(0)});
+    auto gather_m = wrap_type<ov::opset8::Gather>({reshape2_m->output(0), any_input()->output(0), any_input()->output(0)});
+    auto transpose3_m = wrap_type<ov::opset1::Transpose>({gather_m->output(0), any_input()->output(0)});
+    auto final_concat_m1 = wrap_type<ov::opset1::Concat>({dense_feature_m->output(0), transpose3_m->output(0)});
+
+    auto reshape3_m = wrap_type<ov::opset1::Reshape>({gather_m->output(0), any_input()->output(0)});
+    auto transpose4_m = wrap_type<ov::opset1::Transpose>({reshape3_m->output(0), any_input()->output(0)});
+    auto reshape4_m = wrap_type<ov::opset1::Reshape>({transpose4_m->output(0), any_input()->output(0)});
+    auto final_concat_m2 = wrap_type<ov::opset1::Concat>({dense_feature_m->output(0), reshape4_m->output(0)});
+
+    matcher_pass_callback callback = [=](Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
```

Comment:
Shouldn't we also check here concatenation axis, ranks or even particular shapes? Or the impl on CPU plug-in level will correctly cover all cases?

Reply:
The input feature shapes are checked inside Interaction `validate_and_infer_types`, to see whether the conversion is compatible.

Reply:
`validate_and_infer_types` is called after InteractionNode creation. If validation fails - it terminates the whole model compilation. My point is: we shouldn't create InteractionNode  for invalid shapes to avoid that,

Reply:
add input shapes equality check in transformation pass in 0a858793244bc2da0c635de34de8cb9248a9f67b